tests/compile-fail/shim_arg_size.rs

```diff
@@ -0,0 +1,17 @@
+#![feature(rustc_private)]
+
+fn main() {
+    extern "C" {
+        // Use the wrong type(ie. not the pointer width) for the `size`
+        // argument.
+        #[cfg(target_pointer_width="64")]
+        fn malloc(size: u32) -> *mut std::ffi::c_void;
+
+        #[cfg(target_pointer_width="32")]
+        fn malloc(size: u16) -> *mut std::ffi::c_void;
```

Comment:
> `size: u16`

How about `size: u64`? These cases test data_size < target_size, whereas we should test data_size != target_size.